src/components/overview/OverviewMain.jsx

```diff
@@ -69,6 +76,70 @@ class overviewMain extends React.Component {
     }
   }
 
+  //this function is attached to the thumbnail images in the image carousel and
+  //changes the selected image when these images are clicked.
+  //My logic is needlessly complex--I'm using an index for my current image when I should be
+  //using the url. Will change perhaps in the future
+  changeImageOnThumbnailClick(newThumbnailUrl) {
+    for (let i = 0; i < this.state.images.otherImagesInStyle.length; i++) {
+      if (
+        this.state.images.otherImagesInStyle[i].thumbnail_url ===
+        newThumbnailUrl
+      ) {
+        this.setState({
+          images: {
+            currentImage: i,
+            otherImagesInStyle: this.state.images.otherImagesInStyle,
+            currentThumbnailRow: this.state.images.currentThumbnailRow
+          }
+        });
+        break;
+      }
+    }
+  }
+
+  //this function changes the displayed image when you click the arrow
+  onImageArrowClick(direction) {
+    if (direction === 'left') {
+      this.setState({
+        images: {
+          currentImage: this.state.images.currentImage - 1,
+          otherImagesInStyle: this.state.images.otherImagesInStyle,
+          currentThumbnailRow: this.state.images.currentThumbnailRow
+        }
+      });
+    } else if (direction === 'right') {
+      this.setState({
+        images: {
+          currentImage: this.state.images.currentImage + 1,
+          otherImagesInStyle: this.state.images.otherImagesInStyle,
+          currentThumbnailRow: this.state.images.currentThumbnailRow
+        }
+      });
+    }
+  }
+
+  //this function changes the displayed thumbnail images when you click the arrow
+  onNavArrowClick(direction) {
+    if (direction === 'left') {
+      this.setState({
+        images: {
+          currentImage: this.state.images.currentImage,
+          otherImagesInStyle: this.state.images.otherImagesInStyle,
+          currentThumbnailRow: this.state.images.currentThumbnailRow - 1
```

Comment:
I see that images.currentThumbnailRow is changing, but are the other two set state changes redundant? Seem to be setting the state to its current value. Applies above and below as well, I think